packages/react/core/src/features/useWalletFeature.ts

```diff
@@ -0,0 +1,30 @@
+import {
+    WalletStandardError,
+    safeCaptureStackTrace,
+    WALLET_STANDARD_ERROR__FEATURES__WALLET_FEATURE_UNIMPLEMENTED,
+} from '@wallet-standard/errors';
+
+import type { WalletHandle } from '../WalletHandleRegistry_INTERNAL_ONLY_NOT_FOR_EXPORT.js';
+import { getWalletForHandle_DO_NOT_USE_OR_YOU_WILL_BE_FIRED } from '../WalletHandleRegistry_INTERNAL_ONLY_NOT_FOR_EXPORT.js';
+
+/**
+ * Returns the feature object from the Wallet Standard `Wallet` that underlies a `WalletHandle`. In
+ * the event that the wallet does not support the feature, a `WalletStandardError` will be thrown.
+ */
+export function useWalletFeature<TWalletHandle extends WalletHandle>(
+    walletHandle: TWalletHandle,
+    featureName: TWalletHandle['features'][number]
+): unknown {
+    const wallet = getWalletForHandle_DO_NOT_USE_OR_YOU_WILL_BE_FIRED(walletHandle);
+    if (!(featureName in wallet.features)) {
+        const err = new WalletStandardError(WALLET_STANDARD_ERROR__FEATURES__WALLET_FEATURE_UNIMPLEMENTED, {
+            featureName,
+            supportedChains: [...wallet.chains],
+            supportedFeatures: Object.keys(wallet.features),
+            walletName: wallet.name,
+        });
+        safeCaptureStackTrace(err, useWalletFeature);
+        throw err;
+    }
+    return wallet.features[featureName];
```

Comment:
What's the reason for this to be a hook instead of just a plain function? Does it automatically get re-computed if the wallet's features change?

Reply:
You're totally right about this. I'm going to move these all into `/ui` and make them not hooks. That way other libraries can build on them.